packages/jsii/lib/assembler.ts

```diff
@@ -45,6 +45,13 @@ export class Assembler implements Emitter {
   private readonly _submoduleMap = new Map<ts.Symbol, ts.Symbol>();
   private readonly _submodules = new Map<ts.Symbol, SubmoduleSpec>();
 
+  private readonly projectVersion: string;
+  // Map of nodes to types for exported class types. Used by the RuntimeTypeInfoInjector
+  private readonly classTypesMap = new Map<
```

Comment:
So after getting this working, I started a minor refactor to do just that. I ran into some issues and ended up dropping it. Most notably, just taking in and/or updating a Map with a signature like below didn't work because I wasn't able to get a reference to a `ts.ClassDeclaration` prior to running the transformation that would be `===` to the same node during the transformation. I tried a few things with the signatures and tests to hack around it, but wasn't super happy with the result. I'll give it another go and see if I can come up with something that looks prettier.